### PR TITLE
Add Cypress integration with Donation widget

### PR DESCRIPTION
**Pull request by orlando**
**What:** Add Cypress integration with Donation widget.

A test for the membership widget will not be added in this PR.

**Why:** Related to https://app.asana.com/0/1168997577035609/1199605570263293. This aims to reduce the issues we are getting from the membership and union widget.

**How:**

- Setup cypress using the [cypress-rails](https://github.com/testdouble/cypress-rails) gem
- Create test routes to render the widget for the cypress tests to run against

**Media:**

This is from the Cypress Dashboard. These recordings are really useful and helped me debug why e2e was failing.

https://user-images.githubusercontent.com/849872/104079911-d43e9380-51ea-11eb-96a1-90571442dce0.mp4

### CODE REVIEW

**Review by arathjz — APPROVED** [review]

File: cypress/integration/donation.spec.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  beforeEach(() => {
+    cy.request('/cypress_rails_reset_state')
+  })
+
+  it('renders donation widget', () => {
+    cy.visit('/test/widget/donation')
+    cy.contains('DONATE', { matchCase: false })
+  })
+
+  it('makes a donation', () => {
```
Comment (duranmla) [inline cypress/integration/donation.spec.js:13]: How this test is different from the ones we already have?
It seems you are using a mock backend endpoint too, or I am wrong?

**Review by duranmla — COMMENTED** [review]
I don't have anything against this approach but I am concern that we are duplicating efforts. Are we hitting the actual endpoint here? I can't tell :/

File: cypress/integration/donation.spec.js (hunk 1 above)
Comment (orlando) [reply cypress/integration/donation.spec.js:13]: Yeah, we are using the actual endpoint, this is configured here. 

This is an integration test, so if you are mocking the endpoint probably the tests will look the same (actually I don't know if we should port these tests over). There's no easy way to set up test cases without having access to the backend.

https://github.com/debtcollective/membership/pull/373/files#diff-3e7704557ff348aeb70b9950524b0d1924b85cbc6b567d2fe209596d22de1b4dR18-R22

**Comment by orlando** [comment]
I'm going to merge this to continue working with this base on another branch. @duranmla if you have any other comments please let me know 🙏

File: cypress/integration/donation.spec.js (hunk 1 above)
Comment (duranmla) [reply cypress/integration/donation.spec.js:13]: Then we are doing a proper e2e not an integration test.
- **Integration:** Verify that several units work together in harmony. Mocking a little as possible often times requests
- **E2E:** Typically these will run the entire application (both frontend and backend) and your test will interact with the app just like a typical user would. Sometimes called "functional testing" or e2e.

https://kentcdodds.com/blog/unit-vs-integration-vs-e2e-tests

**Pull request merged**